pkgs/development/haskell-modules/patches/readline-fix-for-cabal-3.patch

```diff
@@ -0,0 +1,8 @@
+--- a/Setup.hs	2021-02-04 14:01:09.557970245 +0100
```

Comment:
In general we try not to carry around patches in nixpkgs if it is possible to fetch them from upstream.

Can you fetch this from upstream?

Reply:
Doesn't exist upstream as far as I am aware. The alternative would be to build this with Cabal 2 instead of Cabal 3. I thought it was using Cabal the build system, but it seems it is only using Cabal the library, so it might be possible to pin the dependency instead.

Reply:
Could create an issue upstream (or send a PR fixing this)?

Reply:
I am planning to, but at the moment it is more urgent to me to get this fixed in nix, so I fixed it for myself with this change. If you prefer to wait until upstream fixes the problem that is fine for me.

Reply:
If you want this fixed in Nix, I'd like you to be able to link to the upstream issue/PR so that it is obvious why the patch is needed (and we can keep track of whether or not we can ever remove the patch).

Also, if you send a PR upstream, you can sometimes pull the patch directly from that, so we don't have to carry it around in nixpkgs.

Reply:
So do you want me to close this issue until I have more time to figure out how to fix it without a patch? I don't think there is a problem upstream, maybe pinning the dependencies, but I am not sure. This started failing with a change in nixpkgs, I suspect that moving from Cabal 2 to Cabal 3, but I haven't verified it.

Reply:
>  I don't think there is a problem upstream

Hmm, I'm not sure what could be going on here then.  If this isn't a problem upstream, then it also shouldn't be a problem here in nixpkgs.

> This started failing with a change in nixpkgs, I suspect that moving from Cabal 2 to Cabal 3, but I haven't verified it.

If this is failing upstream when building with Cabal 3, then it is arguably a problem upstream.  I'd prefer that you send a proper fix upstream (or at least create an issue that the Nixpkgs maintainers can track), rather than just hacking around it here in nixpkgs.

Reply:
I guess we could say that the issue is that the package doesn't limit the version of Cabal here: 

http://hackage.haskell.org/package/readline-1.0.3.0/src/readline.cabal

So a proper non invasive fix would be to edit the cabal file and state that Cabal version has to be lower than 3. Will nixpkgs build it with Cabal 2 if I changed that? I got the impression that the Cabal version is not configurable, but I may be wrong abou tthat.

I can't really find what upstream is for this package, I haven't seen any link to the source code other than the package uploaded to hackage.

Reply:
Ah, that is a tricky situation!

The email address in the cabal file is libraries@haskell.org, so I guess you could try to contact that mailing list, but eh, that's quite annoying to do.

In this case, how about we just go ahead and merge this in like you have it, since it doesn't sound like there is a very good solution.

Reply:
At least I can leave a comment in the nix config file, if that helps.

Reply:
Or, let me shoot them an email. I'll get back to you with the outcome of that.